torchinductor/ir.py

```diff
@@ -1893,7 +1893,11 @@ def create(cls, a, b):
             size=[b3, m, n],
         ).as_fixed()
 
-        if b3 == 1:
+        if (
+            b3 == 1
+            and is_contiguous_storage_and_layout(a)
```

Comment:
this doesn't look right, MatrixMultiply should be able to handle transposed inputs

Reply:
The problem is because `View.create` does not return a `ReinterpretView` because of the checking at https://github.com/pytorch/torchdynamo/blob/044e8bd85757e6cb46ff29cbe4c56227809ecdea/torchinductor/ir.py#L695, and thus later failed an assertion at https://github.com/pytorch/torchdynamo/blob/044e8bd85757e6cb46ff29cbe4c56227809ecdea/torchinductor/ir.py#L1570.

I think making a special case in `View.create` might work. @jansel , any suggestion?

Reply:
Hrm, in this case perhaps we should be treating it as contiguous, and ignore size=1 dims in contiguous checks.  The `bs=1` doesn't actually matter.

Reply:
`size=1` is already not checked, but one (or both) of the 2d matrices resulting from removing batch dimension might not be contiguous, and thus `View.create` returns  `View` and not `ReinterpretView`. I think we can return `ReinterpretView` in more cases by specifying proper strides in the `new_layout`

Reply:
I think we should extend SqueezeView.create to accept a specific dimension, and use `Squeezeview.create` instead of `View.create` below.